impl/implstream/kafka.go

```diff
@@ -84,3 +88,12 @@ func (c *kafkaConsumer) Close() error {
 	err := c.reader.Close()
 	return errors.E(op, err)
 }
+
+func getMetadataFromMessage(msg kafka.Message) map[string][]byte {
+	meta := map[string][]byte{}
+	for _, header := range msg.Headers {
+		meta[header.Key] = header.Value
+	}
+	meta[goduck.KeyMetadata] = msg.Key
```

Comment:
It's not much of a problem now but I think this kind of code can be a little dangerous in the future if  someone puts a key on the headers that conflicts with this key.

Reply:
yep, that's why the attribute starts with `goduck_`.
I think a better solution would be to have a separate map of special attributes, or even individual variables. I didn't want the second solution since to all queues will have "key"

Reply:
The problem is if I manually put a new header on kafka like goduck_key=whatever this will be silently overridden by this code.
Because this is an attribute, it will get mixed with my own code attributes. Suppose our backup flow stores the message and all it's attributes and we write a code to restore those messages but they are published under a new key. We would loose the previous key information.
I think it's a small issue, but when we start really using this we could end up with some unforeseen case in which we loose some attributes.

Reply:
I think this problem is a bit unavoidable. Even if we treat it correctly with Go, we're still constrained by how the technology works with Metadata. Worst case would be guaranteed no conflicts in code, but then they reduce to the same key in the Metadata, out of the programmers view.

A work-around would be to hide the Metadata into a goduck type, and then return errors on repeated keys or go-duck-special-keys being used.